src/main/java/uk/gov/companieshouse/acsp/service/FilingsService.java

```diff
@@ -98,20 +98,28 @@ private void buildFilingStatus(FilingApi filing) {
     filing.setCost(costAmount);
   }
 
-  private void buildPresenter(Map<String, Object>data, AcspDataDto acspDataDto) {
+  private void buildPresenter(Map<String, Object> data, AcspDataDto acspDataDto) {
     var presenter = new Presenter();
 
-    presenter.setFirstName(Optional.ofNullable(acspDataDto.getFirstName())
-            .map(String::toUpperCase).orElse(null));
-    presenter.setLastName(Optional.ofNullable(acspDataDto.getLastName())
-            .map(String::toUpperCase).orElse(null));
+    if (acspDataDto.getApplicantDetails() != null) {
+      presenter.setFirstName(Optional.ofNullable(acspDataDto.getApplicantDetails().getFirstName())
+              .map(String::toUpperCase).orElse(null));
+      presenter.setLastName(Optional.ofNullable(acspDataDto.getApplicantDetails().getLastName())
+              .map(String::toUpperCase).orElse(null));
+    } else {
+      presenter.setFirstName(null);
```

Comment:
Do you have to set these to null explicitly?

Reply:
No